### ISSUE TITLE
OSPF collector shouldn't report neighbors on passive interfaces

### ISSUE THREAD

**Issue by matejv** [tynany/frr_exporter#31]
I think if an OSPF interface is configured as passive, frr_exporter should not report neighbors and adjacencies metrics for that interface.

A passive interface means OSPF protocoll will not run on that interface, but the router will still generate network LSAs for address configured on interface.

If an interface is configured as passive it will have the key `"timerPassiveIface":true`:

```json
    "peerlink.4094":{
      "ifUp":true,
      "ifIndex":62,
      "mtuBytes":9000,
      "bandwidthMbit":2000,
      "ifFlags":"<UP,BROADCAST,RUNNING,MULTICAST>",
      "ospfEnabled":true,
      "ipAddress":"169.254.1.1",
      "ipAddressPrefixlen":30,
      "ospfIfType":"Broadcast",
      "localIfUsed":"169.254.1.3",
      "area":"0.0.0.75 [Stub]",
      "routerId":"10.200.1.222",
      "networkType":"BROADCAST",
      "cost":50,
      "transmitDelaySecs":1,
      "state":"DR",
      "priority":1,
      "timerMsecs":10000,
      "timerDeadSecs":40,
      "timerWaitSecs":40,
      "timerRetransmitSecs":5,
      "timerPassiveIface":true,
      "nbrCount":0,
      "nbrAdjacentCount":0
    }
```

If an interface is not passive, that key does not exist:

```json
    "peerlink.4094":{
      "ifUp":true,
      "ifIndex":62,
      "mtuBytes":9000,
      "bandwidthMbit":2000,
      "ifFlags":"<UP,BROADCAST,RUNNING,MULTICAST>",
      "ospfEnabled":true,
      "ipAddress":"169.254.1.1",
      "ipAddressPrefixlen":30,
      "ospfIfType":"Broadcast",
      "localIfUsed":"169.254.1.3",
      "area":"0.0.0.75 [Stub]",
      "routerId":"88.200.1.222",
      "networkType":"BROADCAST",
      "cost":50,
      "transmitDelaySecs":1,
      "state":"DR",
      "priority":1,
      "mcastMemberOspfAllRouters":true,
      "mcastMemberOspfDesignatedRouters":true,
      "timerMsecs":10000,
      "timerDeadSecs":40,
      "timerWaitSecs":40,
      "timerRetransmitSecs":5,
      "timerHelloInMsecs":6182,
      "nbrCount":0,
      "nbrAdjacentCount":0
    }
```

As it stands now, in both cases, frr_exporter generates metrics like:

```
frr_ospf_neighbor_adjacencies{area="0.0.0.75 [Stub]",iface="peerlink.4094",vrf="default"} 0
frr_ospf_neighbors{area="0.0.0.75 [Stub]",iface="peerlink.4094",vrf="default"} 0
```

This makes very difficult to properly alert on an OSPF interface that lost its neighbor, for example.

**Comment by tynany** [comment]
I agree. A fix for this issue has been released in [v0.1.14](https://github.com/tynany/frr_exporter/releases/tag/v0.2.14). Please validate and let me know. Thanks!

**Comment by matejv** [comment]
On the latest version it works - no metrics for interfaces configured as passive. Thanks!
